src/main/scala/chisel3/util/SparseVec.scala

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
+
+import chisel3._
+import chisel3.experimental.SourceInfo
+import chisel3.util.{log2Up, BitPat, Mux1H}
+import chisel3.util.experimental.decode.{decoder, TruthTable}
+import scala.collection.mutable.{LinkedHashMap, LinkedHashSet}
+
+sealed trait SparseVec[A] {
```

Comment:
I think it can be a Bundle with custom fields for better Verilog

Reply:
@jackkoenig suggested making this a `Record` and having it support writes. While I think this is possible, I'm concerned that it's subtly something different. Specifically, the `SparseVec` currently sets a default value for reading if you read something that isn't in the `Vec`. However, I'm not sure how that should work for writing... are the writes dropped or are writes to values not called out visible?

Regardless, I'll play around with making the internal storage a `Record`.

Reply:
Switched to `Record` in latest version.